remote/analyze_handler.py

```diff
@@ -55,33 +57,79 @@ def preAnalyze(analyze_id, part_number, workspace):
 
     analyze_dir_path = os.path.join(file_path, source_file_without_extension)
 
-    with zipfile.ZipFile(zip_file_path) as zf:
+    with zipfile.ZipFile(zip_file_path) as zip_file:
         try:
-            zf.extractall(analyze_dir_path)
+            zip_file.extractall(analyze_dir_path)
         except Exception:
-            LOGGER.error("Failed to extract all files from the ZIP.")
-
-    command_file_path = os.path.join(
-        analyze_dir_path, 'sources-root', 'build_command')
-
-    with open(command_file_path) as build_action:
+            LOGGER.error('Failed to extract all files from the ZIP.')
+
+        list_of_other_files = ['sources-root/build_command',
+                               'sources-root/file_path',
+                               'sources-root/skipped_file_list']
+
+        if (len(zip_file.namelist()) > 0):
+            LOGGER.info('Load files to Redis: ' % zip_file.namelist())
+            for file_name in zip_file.namelist():
+                if file_name not in list_of_other_files:
+                    LOGGER.info(os.path.join(analyze_dir_path, file_name))
+                    with open(os.path.join(analyze_dir_path, file_name), 'rb') as file:
+                        bytes = file.read()
+                        readable_hash = hashlib.md5(bytes).hexdigest()
+                        try:
+                            REDIS_DATABASE.set(readable_hash, bytes)
+                            LOGGER.info(REDIS_DATABASE.get(readable_hash))
+                        except Exception:
```

Comment:
Will the analysis be successful if some files are missing? Shouldn't we skip any further processing and set some failed flag?